lib/ansible/modules/windows/win_hostname.ps1

```diff
@@ -0,0 +1,57 @@
+#!powershell
+# Copyright: (c) 2018, Ripon Banik (@riponbanik)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# Requires -Module Ansible.ModuleUtils.Legacy
+
+########
+
+function Get-ComputerName() {
+    Get-Content env:computername
+    return
+}
+
+function Set-ComputerName($newComputerName) {
```

Comment:
As mentioned below, use `Rename-Computer` for this instead of messing with WMI and checking the return value.